merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/ConstraintAction.java

```diff
@@ -34,74 +31,78 @@ public ConstraintAction(
+   */
+  public Pair<Integer, Map<ConstraintRecord, Fallible<ConstraintResult, List<? extends Exception>>>> getViolations(
+      final PlanId planId,
+      final Optional<SimulationDatasetId> simulationDatasetId,
+      final boolean force,
+      final HasuraAction.Session userSession
+  ) throws NoSuchPlanException, MissionModelService.NoSuchMissionModelException, SimulationDatasetMismatchException {
     final var plan = this.planService.getPlanForValidation(planId);
-    final Optional<SimulationResultsHandle> resultsHandle$;
-    final SimulationDatasetId simDatasetId;
+
+    // Get a Handle for the Simulation Results
+    final SimulationResultsHandle resultsHandle;
     if (simulationDatasetId.isPresent()) {
-      resultsHandle$ = this.simulationService.get(planId, simulationDatasetId.get());
-      simDatasetId = resultsHandle$
-          .map(SimulationResultsHandle::getSimulationDatasetId)
-          .orElseThrow(() -> new InputMismatchException("simulation dataset with id `"
-                                                        + simulationDatasetId.get().id()
-                                                        + "` does not exist"));
+      resultsHandle = this.simulationService.get(planId, simulationDatasetId.get())
+                                            .orElseThrow(() -> new InputMismatchException(
+                                                "simulation dataset with id `"
+                                                + simulationDatasetId.get().id()
+                                                + "` does not exist"));
     } else {
       final var revisionData = this.planService.getPlanRevisionData(planId);
-      resultsHandle$ = this.simulationService.get(planId, revisionData);
-      simDatasetId = resultsHandle$
-          .map(SimulationResultsHandle::getSimulationDatasetId)
-          .orElseThrow(() -> new InputMismatchException("plan with id "
-                                                        + planId.id()
-                                                        + " has not yet been simulated at its current revision"));
+      resultsHandle = this.simulationService.get(planId, revisionData)
+                                            .orElseThrow(() -> new InputMismatchException(
+                                                "plan with id "
+                                                + planId.id()
+                                                + " has not yet been simulated at its current revision"));
     }
 
-    final var constraintCode = new HashMap<>(this.planService.getConstraintsForPlan(planId));
-    final var constraintResultMap = new HashMap<Constraint, Fallible<?>>();
+    final SimulationDatasetId simDatasetId = resultsHandle.getSimulationDatasetId();
 
-    final var validConstraintRuns = this.constraintService.getValidConstraintRuns(constraintCode, simDatasetId);
+    final var constraintCode = new ArrayList<>(this.planService.getConstraintsForPlan(planId));
```

Comment:
`constraintCode` seems a bit misleading now that it includes both EDSL and JAR constraints. Perhaps `constraintRecords` would be more descriptive?

Reply:
I've renamed it to `constraints`